tensorflow/core/kernels/dml_cwise_ops.cc

```diff
@@ -481,25 +489,53 @@ REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(Minimum, dml::Min(x, y), 8, true,
 // cwise_op_maximum.cc).
 REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(Maximum, dml::Max(x, y), 8, true,
                                        Eigen::half, float, int64)
-REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(SquaredDifference,
+REGISTER_DML_COMPOSITE_BINARY_KERNEL_2(SquaredDifference,
                                        dml::DifferenceSquare(x, y), 8, true,
-                                       Eigen::half, float, int64)
+                                       Eigen::half, float)
+// TODO: Revisit this and consider having a native int64 alternative
+// TFDML #41163316
+REGISTER_DML_COMPOSITE_BINARY_KERNEL_1(SquaredDifference, (x - y) * (x - y), 8,
+                                       false, int64)
 // TODO(b/25387198): A special kernel exists for int32 (see cwise_op_mul1.cc).
 REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(Mul, (x * y), 8, true, Eigen::half,
                                        float, int64)
-REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(Pow, dml::Pow(x, y), 8, true,
-                                       Eigen::half, float, int64)
+REGISTER_DML_COMPOSITE_BINARY_KERNEL_2(Pow, dml::Pow(x, y), 8, true,
+                                       Eigen::half, float)
+// TODO: Revisit this and consider having a native int64 alternative
+// TFDML #41163316
+REGISTER_DML_COMPOSITE_BINARY_KERNEL_1(
+    Pow,
+    dml::Cast(dml::Pow(dml::Cast(x, DML_TENSOR_DATA_TYPE_INT32),
+                       dml::Cast(y, DML_TENSOR_DATA_TYPE_INT32)),
+              DML_TENSOR_DATA_TYPE_INT64),
+    8, false, int64)
 // TODO(b/25387198): A special kernel exists for int32 (see cwise_op_add1.cc).
 REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(Add, x + y, 8, true, Eigen::half, float,
                                        int64)
 // TODO(b/25387198): A special kernel exists for int32 (see cwise_op_add1.cc).
 REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(AddV2, x + y, 8, true, Eigen::half,
                                        float, int64)
-REGISTER_DML_COMPOSITE_BINARY_KERNEL_4(TruncateDiv, x / y, 8, true, uint8,
-                                       uint16, int16, int64)
+REGISTER_DML_COMPOSITE_BINARY_KERNEL_3(TruncateDiv, x / y, 8, true, uint8,
+                                       uint16, int16)
```

Comment:
>uint16, int16 [](http://example.com/codeflow?start=39&length=13)

Should this include uint32 and int32 too? Otherwise it's weird that there's a gap.

Reply:
TruncateDiv doesn't define uint32 for TruncateDiv at the C++ level, and int32 is reserved for CPU.

Reply:
Silly TF.